wallet-api/src/Wallet/API.hs

```diff
@@ -195,6 +201,22 @@ data WalletAPIError =
 instance FromJSON WalletAPIError
 instance ToJSON WalletAPIError
 
+newtype WalletLog = WalletLog { getWalletLog :: [Text] }
+    deriving (Eq, Ord, Show, Generic, Semigroup, Monoid)
+
+instance FromJSON WalletLog
+instance ToJSON WalletLog
+
+-- | Write some information to the log
+logMsg :: MonadWriter WalletLog m => Text -> m ()
+logMsg = tell . WalletLog . pure
+
+-- | WalletAPI with the ability to log messages and throw errors
+type LoggingWallet m = (
```

Comment:
I don't like the name given that it includes errors too. How about `WalletDiagnostics`? I'd also probably not imply `WalletApi` for clarity.

Reply:
Although that's a bit subjective.